Document dedicated limitation regarding MyISAM tables (#1385)
platformsh/platformsh-docs: src/configuration/services/mysql.md

> **note**
>
> Downgrades of MySQL or MariaDB are not supported. Both will update their own datafiles to a new version automatically but cannot downgrade them. If you want to experiment with a later version without committing to it use a non-master environment.

> > Dedicated environments do not support any storage engine other than InnoDB. Tables created using the MyISAM storage engine on dedicated environments will not replicate between cluster nodes.

Reviewer: ```suggestion
>
> Dedicated environments do not support any storage engine other than InnoDB. Tables created using the MyISAM storage engine on dedicated environments will not replicate between cluster nodes.
```

Consult the [MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/charset-mysql.html) for further details.

## Storage Engine

> We recommend using the InnoDB storage engine wherever possible. MyISAM is only properly supported in Grid environments, in dedicated cluster environments there is no replication of MyISAM tables.

Reviewer: ```suggestion
We recommend using the InnoDB storage engine wherever possible. MyISAM is only properly supported in Grid environments. In dedicated cluster environments there is no replication of MyISAM tables.
```

> If MyISAM tables are inadventently created or imported in a dedicated environment they can be easily converted to use the InnoDB storage engine using the following proceedure:

Reviewer: ```suggestion
If MyISAM tables are inadventently created or imported in a dedicated environment they can be converted to use the InnoDB storage engine using the following procedure:
```

```sql
RENAME TABLE <existing> <old>;

> INSERT INTO <existing> SELECT * from <old>;

Reviewer: Wouldn't a rename include data?  This seems... odd.  I think the first is supposed to be a create statement or something, no?

Author: It's a rename. Essentially moving it out of the way and then recreating it via the INSERT INTO so that it gets recreated with the correct, default storage engine and proper replication.
